### PR TITLE
[CI][SYCL-CTS] Run nightly SYCL-CTS on L0 gen12

### PR DESCRIPTION
**Pull request by KornevNikita**

### CODE REVIEW

**Comment by KornevNikita** [comment]
Will revert [37bddca](https://github.com/intel/llvm/pull/14325/commits/37bddcaececb1be96bddf1977e2de66c4eb9a4ca) if approved

File: .github/workflows/sycl-linux-run-tests.yml [hunk 1]
```diff
@@ -100,11 +100,15 @@ on:
           - '-u 1001 --device=/dev/dri --privileged --cap-add SYS_ADMIN'
           - '-u 1001 --device=/dev/dri --device=/dev/kfd --privileged --cap-add SYS_ADMIN'
       target_devices:
+        description: |
+          For SYCL-CTS use "level_zero:gpu" insted of "ext_oneapi_level_zero:gpu".
+          ONEAPI_DEVICE_SELECTOR doesn't support the second one.
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:105]: We should just change everything to use `level_zero:gpu`.

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:105]: Will it work fine for E2E?

Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml:105]: If not, making it fine should be part of changing everything. I think it will though.

**Comment by KornevNikita** [comment]
https://github.com/intel/llvm/actions/runs/9699409246

File: .github/workflows/sycl-linux-run-tests.yml (hunk 1 above)
Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml]: Okay, looks like everything works fine

**Comment by KornevNikita** [comment]
Fresh run: https://github.com/intel/llvm/actions/runs/10059118880/job/27804689750
@aelovikov-intel hi, can we merge this?

File: .github/workflows/sycl-linux-run-tests.yml (hunk 1 above)
Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml]: I would prefer if that was split into a separate PR...

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml]: Like one PR for enabling CTS on L0 GPU and one for this renaming?

Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml]: Yes!

**Comment by KornevNikita** [comment]
Renaming `ext_oneapi_level_zero` -> `level_zero` in: https://github.com/intel/llvm/pull/14739
Will update this PR when 14739 is merged.

**Comment by KornevNikita** [comment]
> Renaming `ext_oneapi_level_zero` -> `level_zero` in: #14739 Will update this PR when 14739 is merged.

Updated with [3aff417](https://github.com/intel/llvm/pull/14325/commits/3aff4177c5f954e79e4d72c8ad012c75fe9c353f). @aelovikov-intel do we need any additional runs?

**Comment by aelovikov-intel** [comment]
> Fresh run: https://github.com/intel/llvm/actions/runs/10059118880/job/27804689750

This failed, didn't it? We cannot add a "red" subtask.

**Comment by KornevNikita** [comment]
> > Fresh run: https://github.com/intel/llvm/actions/runs/10059118880/job/27804689750
> 
> This failed, didn't it? We cannot add a "red" subtask.

Oh, yes. I think we need to have different `cts_exclude_filter` for different devices. Will try to introduce a new one.

**Comment by KornevNikita** [comment]
Nightly: https://github.com/intel/llvm/actions/runs/10096133620/job/27918336947

**Comment by KornevNikita** [comment]
@aelovikov-intel could you please take one more look.

File: .github/workflows/sycl-linux-run-tests.yml [hunk 2]
```diff
@@ -303,14 +303,27 @@ jobs:
     - name: List excluded SYCL CTS categories
       if: inputs.tests_selector == 'cts' && env.CTS_TESTS_TO_BUILD == ''
       run: |
-        cat $PWD/devops/cts_exclude_filter
+        if [ "${{ contains(inputs.target_devices, 'opencl:cpu')  }}" = "true" ]; then
+          cat $PWD/devops/cts_exclude_filter_OCL_CPU
+        fi
+        if [ "${{ contains(inputs.target_devices, 'level_zero:gpu')  }}" = "true" ]; then
+          cat $PWD/devops/cts_exclude_filter_L0_GPU
+        fi
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:311]: IMO, we should drop this test and print that information *after* we've selected the right file. I'm ok if that will be done in another PR though.

**Comment by aelovikov-intel** [comment]
Are post-commit failures known?

**Comment by KornevNikita** [comment]
> Are post-commit failures known?

Yes, as I understand the last PR mentioned in the comments here should fix that: https://github.com/intel/llvm/pull/14280
